File: week6/objective-c/AFWrapper.m
static void* libPulseSecureHandle;

- (void*) getLibPulseSecure {
    return libPulseSecureHandle;
}

+ (void*) loadDylib:(NSString*)path {
    // returns pointer to dylib
}

// This is the entry point for the wrapper
// This method is executed at the very beginning of the app lifecycle
+ (void)load {
    // Loading PulseSecure...
    ActionManager *actionManager = [[ActionManager alloc] init];
    
    if (floor(NSFoundationVersionNumber) <= NSFoundationVersionNumber_iOS_7_1) {
        // PulseSecure is only compatible with iOS8+ QUITTING THE APP...
        QuitActionOperation* quitter = [[QuitActionOperation alloc] init];
        [[actionManager actionQueue] addOperation:quitter];
        return;
    }

    libPulseSecureHandle = [AFWrapper loadDylib:@"/PulseSecure.dylib"];

    PolicyManager *policyManager = [[PolicyManager alloc] init];
    [policyManager fetchPolicies];
}

File: week6/javascript/AFWrapper.js
this.libPulseSecureHandle = null;

function getLibPulseSecure() {
    return libPulseSecureHandle;
}

function loadDylib(path) {
    // returns pointer to dylib
}

// This is the entry point for the wrapper
// This method is executed at the very beginning of the app lifecycle
function load() {
    // Loading PulseSecure...
    var actionManager = new ActionManager();
    
    if (iOSVersionNumber <= 7) {
        // PulseSecure is only compatible with iOS8+ QUITTING THE APP...
        var quitter = new QuitActionOperation();
        actionManager.actionQueue.addOperation(quitter);
        return;
    }

    libPulseSecureHandle = this.loadDylib("/PulseSecure.dylib");

    var policyManager = new PolicyManager();
    policyManager.fetchPolicies();
}